### PR TITLE
IIP-17/ Feature. Download and attach maven artifacts' javadocs and sources to related …

### PR DESCRIPTION
**Pull request by viktor-yengovatov**
…extensions.

Related artifacts list is based on `external-dependencies.xml` files, artifacts with "-javadoc.jar" or with "-sources.jar" suffixes.

Minor fix for ordering elements in `*.iml` files - more predicted, sorted or ordered elements allows compare and commit `*.iml` easier.

Signed-off-by: Viktors Jengovatovs <viktors.jengovatovs@pearlconsulting.no>

Signed-off-by: Your Real Name your.email@email.com

### CODE REVIEW

File: src/com/intellij/idea/plugin/hybris/settings/HybrisApplicationSettings.java [hunk 1]
```diff
@@ -106,6 +106,12 @@
     @PropertyName("followSymlink")
     private boolean followSymlink = true;
 
+    @PropertyName("withMavenSources")
+    private boolean withMavenSources = true;
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/settings/HybrisApplicationSettings.java]: Please, leave it as _false_.

File: src/com/intellij/idea/plugin/hybris/settings/HybrisApplicationSettings.java [hunk 2]
```diff
@@ -106,6 +106,12 @@
...
+
+    @PropertyName("withMavenJavadocs")
+    private boolean withMavenJavadocs = true;
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/settings/HybrisApplicationSettings.java]: Please, leave it as _false_.

File: src/com/intellij/idea/plugin/hybris/settings/HybrisApplicationSettings.java (hunk 1 above)
Comment (ghost) [reply src/com/intellij/idea/plugin/hybris/settings/HybrisApplicationSettings.java]: We need to give users a chance to choose before sources and docs importing.

File: src/com/intellij/idea/plugin/hybris/settings/HybrisApplicationSettingsForm.java [hunk 3]
```diff
@@ -57,6 +57,8 @@
     private JCheckBox defaultPlatformInReadOnly;
     private JTextField groupPlatformTextField;
     private JCheckBox followSymlink;
+    private JCheckBox withMavenSources;
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/settings/HybrisApplicationSettingsForm.java:60]: Could we add these checkboxes to the project import wizard?

**Review by ghost — COMMENTED** [review]
Hello Viktors, thanks for your contribution.

File: src/com/intellij/idea/plugin/hybris/settings/HybrisApplicationSettingsForm.java (hunk 3 above)
Comment (viktor-yengovatov) [reply src/com/intellij/idea/plugin/hybris/settings/HybrisApplicationSettingsForm.java:60]: Done

**Pull request merged**